sympy/polys/rootoftools.py

```diff
@@ -394,6 +394,20 @@ def free_symbols(self):
         # symbols
         return set()
 
+    def _eval_power(self, other):
+        if not isinstance(other, (Integer, int)):
+            return None
+        d = degree(self.expr)
+        if other < d:
+            return None
```

Comment:
It would also be possible to handle negative powers when the RootOf poly is irreducible over QQ:
```python
In [6]: r = RootOf(z**3 + z + 29, 0)

In [7]: r
Out[7]: 
       ⎛ 3            ⎞
CRootOf⎝z  + z + 29, 0⎠

In [8]: e = 1/r

In [9]: e
Out[9]: 
           1           
───────────────────────
       ⎛ 3            ⎞
CRootOf⎝z  + z + 29, 0⎠

In [10]: N(e)
Out[10]: -0.337396969245337

In [11]: invert(z, z**3 + z + 29)
Out[11]: 
   2     
  z    1 
- ── - ──
  29   29

In [12]: invert(z, z**3 + z + 29).subs(z, r)
Out[12]: 
                         2     
         ⎛ 3            ⎞      
  CRootOf⎝z  + z + 29, 0⎠    1 
- ──────────────────────── - ──
             29              29

In [13]: invert(z, z**3 + z + 29).subs(z, r).n()
Out[13]: -0.337396969245337
```